File: million/million.py
from threading import Thread

X = 0

def add_500k():
    global X
    for _ in range(500_000):
        X += int(1)

thread1 = Thread(target=add_500k)
thread1.start()
thread2 = Thread(target=add_500k)
thread2.start()
thread1.join()
thread2.join()
print(X)

File: million/million.rs
use std::sync::Mutex;

static X: Mutex<i32> = Mutex::new(0);

fn add_500k() {
    for _ in 0..500_000 {
        *X.lock().unwrap() += 1;
    }
}

fn main() {
    let thread1 = std::thread::spawn(add_500k);
    let thread2 = std::thread::spawn(add_500k);
    thread1.join().unwrap();
    thread2.join().unwrap();
    println!("{}", X.lock().unwrap());
}
